fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java

```diff
@@ -0,0 +1,128 @@
+package com.github.dedis.popstellar.model.network.method.message.data.digitalcash;
+
+import com.github.dedis.popstellar.utility.security.Hash;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The transaction object
+ */
+public class Transaction {
+  @SerializedName(value = "version")
+  private final int version; // The version of the transaction inputs
+
+  @SerializedName(value = "inputs")
+  private List<Input> inputs; // [Array[Objects]] array of output transactions to use as inputs
+
+  @SerializedName(value = "outputs")
+  private List<Output> outputs; // [Array[Objects]] array of outputs from this transactions
+
+  @SerializedName("lock_time")
+  private final long lockTime; // LockTime
+
+  /**
+   * Transaction constructor
+   * @param version The version of the transaction inputs
+   * @param inputs [Array[Objects]] array of output transactions to use as inputs
+   * @param outputs [Array[Objects]] array of outputs from this transactions
+   * @param lockTime TimeStamp
+   */
+  public Transaction(int version, List<Input> inputs, List<Output> outputs, long lockTime) {
+    this.version = version;
+    this.inputs = Collections.unmodifiableList(inputs);
+    this.outputs = Collections.unmodifiableList(outputs);
```

Comment:
Typically, you would also defensively copy the list to avoid aliasing. That's a clean code thing, not a bug per se.

Reply:
I am not sure to understand what you mean ? You think that instead of doing an unmodifiableList I should do a defensive copy ? 
I've looked for other json array implementation in the code (for instance ElectionResult or ElectionResultQuestion) but none of them does a defensive copy. 😅

Reply:
You should do both.

Aliasing means there exists more than one pointer to the same object. In this case, you could have this:
```java
List<Inputs> aliased = ...;
var tx = new Transaction(..., aliased, ...);
aliased.add(...); // whoops
tx.getInputs(); // tx has been sneakily modified
```
So if we want immutability, we should defensive copy.

That said, feel free to align with the rest of the codebase and leave this concern out of scope.